src/test/java/org/eolang/speco/SpecoTest.java

```diff
@@ -0,0 +1,165 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.SystemUtils;
+import org.eolang.jucs.ClasspathSource;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.io.CleanupMode;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Packs tests.
+ *
+ * @since 0.0.1
+ */
+class SpecoTest {
+
+    @Tag("fast")
+    @ParameterizedTest
+    @ValueSource(strings = {"simple"})
+    public void convertsFromXmir(
+        final String title,
+        @TempDir(cleanup = CleanupMode.NEVER) final Path temp) throws IOException {
+        final Path base = Path.of(
+            "src", "test", "resources",
+            "org", "eolang", "speco",
+            "xmir", title
+        );
+        new Speco(base.resolve("in"), temp, false).exec();
+        final Path reference = base.resolve("out");
+        for (final Path path : Files.newDirectoryStream(reference)) {
+            MatcherAssert.assertThat(
+                String.format(
+                    "Files %s in %s and %s are different",
+                    path.getFileName(),
+                    temp,
+                    reference
+                ),
+                Files.readAllLines(temp.resolve(path.getFileName())),
+                Matchers.equalTo(
+                    Files.readAllLines(path)
+                )
+            );
+        }
+    }
+
+    @Disabled
+    @Tag("fast")
+    @ParameterizedTest
+    @ClasspathSource(value = "org/eolang/speco/packs", glob = "**.yaml")
+    public void convertsFromEo(
+        final String pack,
+        @TempDir(cleanup = CleanupMode.NEVER) final Path temp) throws IOException {
+        final Map<String, Object> script = new Yaml().load(pack);
+        MatcherAssert.assertThat(
+            "Unexpected transformation result",
+            Files.readString(SpecoTest.runSpeco(script, temp).resolve("app.eo")),
+            Matchers.equalTo(
+                script.get("after").toString()
+            )
+        );
+    }
+
+    @Tag("slow")
+    @ParameterizedTest
+    @ClasspathSource(value = "org/eolang/speco/packs", glob = "**.yaml")
+    public void compilesFromEo(
+        final String pack,
+        @TempDir(cleanup = CleanupMode.NEVER) final Path temp) throws IOException {
+        final Map<String, Object> script = new Yaml().load(pack);
+        MatcherAssert.assertThat(
+            "Unexpected execution result",
+            SpecoTest.exec(SpecoTest.runSpeco(script, temp).toString()),
+            Matchers.equalTo(
+                script.get("result").toString().split("\\r?\\n")
```

Comment:
@advasileva Maybe we can use `System.lineSeparator()` here instead?

Reply:
@volodya-lombrozo Yes, done

Reply:
@advasileva Thanks